### PR TITLE
Release/v0.1.3

### LINKED ISSUE #528: generalSettings.startOfWeek throwing an error
The following query: 

```
query {             
    generalSettings {
        dateFormat
        description
        language
        startOfWeek
        timeFormat
        timezone
        title
    }
}
```

Returns an error: 

```
{
      "debugMessage": "Int cannot represent non 32-bit signed integer value: (empty string)",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 6,
          "column": 5
        }
      ],
      "path": [
        "generalSettings",
        "startOfWeek"
      ]
    }
  ]
```

### LINKED ISSUE #578: WPInputObjectType::construct doesn't call prepare_fields
I'm not sure if there is a new way to register input fields with the type registry implementation, so I'm testing registering an input type "lang" to filter posts by language. The following snippet worked prior to 0.1.0. To reproduce add this snippet as a plugin

```
add_filter( 'graphql_input_fields', function( $fields ) {
	$fields['lang'] = [
		'type' => \WPGraphQL\Types::string(),
		'description' => 'Select posts by Polylang language',
	];
	$fields['allLanguages'] = [
		'type' => \WPGraphQL\Types::boolean(),
		'description' => 'Show posts regardless of the Polylang language',
	];
	return $fields;
} );
```

Then run a query like
```
query Posts($lang: String) {
  posts(where: {lang: $lang}) {
    edges {
      node {
        title
      }
    }
  }
}
---
{
  "lang": "en"
}
```
See https://gist.github.com/epeli/87909f92b19a216dc7f5b2b6c204ad31

While digging into this I found in ./src/Type/WpInputObjectType.php

$config['fields'] was an array prior to 0.1.0 so this if condition prevents prepare_fields() from being called.

```
	public function __construct( $config = [] ) {
		if ( ! empty( $config['fields'] ) && is_array( $config['fields'] ) ) {
			$config['fields'] = self::prepare_fields( $config['fields'], $config['name'], $config );
		}
		parent::__construct( $config );
	}
...
```

A print_r of $config['fields'] here gives me

```
Closure Object
(
    [static] => Array
        (
            [config] => Array
                (
                    [description] => Input for the updateSettings mutation
                    [fields] => Array
...
...
```

I haven't yet been able to figure out what this _should_ be accepting... but figured I'd post this to start a thread... If someone can point me in the direction of what this should be expecting... I can take a crack at trying to update.

### LINKED ISSUE #598: Searching for terms doesn't return child terms in results
Searching for terms doesn't return child terms in the results. 

Here, you can see the term "child category" 

screen shot 2018-12-04 at 11 23 42 am

And when the following query is executed, I would expect to see that term in the results, but it's not. 

```
{
  categories(where: {search: "child"}) {
    nodes {
      id
      categoryId
      name
    }
  }
```

This is due to the default query arg for root TermObjectConnections being set to `parent => 0`

We should unset the `parent` query arg for search requests to make sure searches account for child terms as well.

### PR DESCRIPTION
**Pull request by jasonbahl**
# Release Notes

## New Features
- added `resetUserPassword` mutation – thanks @kellenmace 
- added `sendPasswordResetEmail` mutation – thanks @kellenmace 

## Bug Fixes
- Update logo in README.md
- fixed #528 (generalSettings.startOfWeek throwing an error) – thanks @davidatwhiletrue
- fixes #578, restores the `graphql_input_fields` filter – thanks @chriszarate 
- fixes #598, allows search results of Terms and Posts to include children (default was only objects with no parent).

### CODE REVIEW

File: src/Mutation/SendPasswordResetEmail.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+				if ( ! $user_data ) {
+					throw new UserError( self::get_user_not_found_error_message( $input['username'] ) );
+				}
+				$key = get_password_reset_key( $user_data );
+				if ( is_wp_error( $key ) ) {
+					throw new UserError( __( 'Unable to generate a password reset key.', 'wp-graphql' ) );
+				}
+				$subject    = self::get_email_subject( $user_data );
+				$message    = self::get_email_message( $user_data, $key );
+				$email_sent = wp_mail( $user_data->user_email, wp_specialchars_decode( $subject ), $message );
```
[the remark this answers is not in the document]
Comment (jasonbahl) [reply src/Mutation/SendPasswordResetEmail.php]: that's a good idea. 🤔

**Comment by jasonbahl** [comment]
@CodeProKid do you consider either of your comments blockers that should be addressed now? or issues to address later?

**Comment by CodeProKid** [comment]
@jasonbahl definitely not the second one. I'm on the fence about the first one though. Sending a lot of emails from the server could definitely take down a site. Then again querying the hell out of a complicated query could do just the same, so I'm not sure.

**Comment by CodeProKid** [comment]
Thinking about this a little more, if core does _something_ to limit the amount of PW reset requests, then that should be a blocker for this, as I would expect it to behave the same way. If it doesn't it can go out as is, with issues to address in the future 👍

**Comment by jasonbahl** [comment]
@CodeProKid sounds good. I'll check out how core is doing it and follow-up.

**Pull request merged**